### DISCUSSION TITLE
How can I use `currentElement` in `useForwardExpose`?

### DISCUSSION THREAD

**Discussion by appano1** [unovue/reka-ui#1143] [Q&A]
When I looked into the source, I saw `currentElement` handles only the case when `currentRef.value` is `ComponentPublicInstance`. It doesn't handle when the `current.value` itself is an `Element`.

Additionally, it just ignores the type when itself is `null` or `undefined`.

https://github.com/radix-vue/radix-vue/blob/c157cb820034f5fef25b3d111841c617bb49642f/packages/radix-vue/src/shared/useForwardExpose.ts#L9-L13

Are they intended or bugs? I'm quite confused because other components that uses `currentElement` don't care about the empty case. Is it fine to ignore the empty case and should I use it only when the component is an instance of `ComponentPublicInstance`?

**Comment by zernonia** [comment]
@appano1  All of that being handled by `unrefElement`, which support `ComponentPublicInstance` and `Element`

https://github.com/vueuse/vueuse/blob/main/packages/core/unrefElement/index.ts

**Reply by appano1** [reply]
I've already tested but it makes an error, because when currentRef.value is `Element`, $el will not exist. Therefore `$el.nodeName` will makes an error.

```
Uncaught TypeError: Cannot read properties of undefined (reading 'nodeName')
```

I assume this is a bug. This is simple [example](https://stackblitz.com/edit/vitejs-vite-axokhr?file=src%2Fcomponents%2FHelloWorld.vue).

And what about the type of currentElement? I think it needs to be `HTMLElement | SVGElement | null | undefined`.

**Reply by appano1** [reply]
@zernonia, if you think this needs to be changed, I would be happy to create a PR for it.
